api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -261,6 +261,8 @@ message Cluster {
       // :ref:`keys<envoy_api_field_config.cluster.v3.Cluster.LbSubsetConfig.LbSubsetSelector.keys>` are allowed, but
       // `fallback_keys_subset` cannot be equal to `keys`.
       repeated string fallback_keys_subset = 3;
+
+      bool single_host_per_subset = 4;
```

Comment:
Sure, this looks fine, maybe add some description that this is a more efficient algorithm. Should this be in the subset selector or parent message?

Reply:
I wasn't sure if we eventually wanted to support some selectors being in this mode and some not, so I put it here. But I'm fine putting it in the parent if we think that's better.

Reply:
If your optimization can sensibly work with a mixture, sure, why not provide the flexiblity. But if it implies that they all must be in this mode, then makes sense to move it to the parent.

Reply:
I think it could be adapted with more work to do both. It would need some work in the code that figures out which selector should be used, but I think it could be done if someone wants to. @snowp or @zuercher do either of you have an opinion?